### PR TITLE
Marathon build and startup parameterized for better management and EE alignment

### PR DESCRIPTION
**Pull request by kensipe**
## High Level Description

What features does this change enable? What bugs does this change fix? High-Level description of how things changed.

## Related Issues

  - [DCOS_OSS-18597](https://jira.mesosphere.com/browse/DCOS-18597)Marathon DCOS packaging.

## Checklist for all PR's

  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:  This is the build process itself... it either builds or not and marathon starts or not.  Either way existing tests will fail if it doesn't work.
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review):   N/A
  - [ ] Test Results: N/A
  - [ ] Code Coverage (if available): N/A

### CODE REVIEW

**Comment by kensipe** [comment]
This is a OPEN change only and will NOT have an EE bump

File: packages/marathon/extra/marathon.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+exec $PKG_PATH/marathon/bin/marathon \
+    -Duser.dir=$MARATHON_USER_DIR \
+    $MARATHON_JVM_ARGS \
+    --master $MARATHON_MESOS_MASTER_URL \
+    --default_accepted_resource_roles $MARATHON_ACCEPTED_ROLES \
+    --mesos_role $MARATHON_MESOS_ROLE \
+    --max_instances_per_offer $MARATHON_MAX_INSTANCES_PER_OFFER \
+    --task_launch_timeout $MARATHON_TASK_LAUNCH_TIMEOUT \
+    --decline_offer_duration $MARATHON_DECLINE_OFFER_DURATION \
```
Comment (timcharper) [inline packages/marathon/extra/marathon.sh]: This seems redundant. The `bin/marathon` script already implicitly converts environment variables to parameters, see https://github.com/mesosphere/marathon/blob/master/project/NativePackagerSettings/extra-defines.bash#L15

Comment (ichernetsky) [reply packages/marathon/extra/marathon.sh]: I concur with Tim. Also, there is https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/Main.scala#L41

File: gen/dcos-config.yaml [hunk 2]
```diff
@@ -884,6 +884,18 @@ package:
     content: |
       MARATHON_ZK=zk://zk-1.zk:2181,zk-2.zk:2181,zk-3.zk:2181,zk-4.zk:2181,zk-5.zk:2181/marathon
       LIBPROCESS_PORT=15101
+      MARATHON_MESOS_MASTER_URL=zk://zk-1.zk:2181,zk-2.zk:2181,zk-3.zk:2181,zk-4.zk:2181,zk-5.zk:2181/mesos
```
Comment (ichernetsky) [inline gen/dcos-config.yaml]: Are these variables going literally into `/etc_master/marathon`?

Comment (kensipe) [reply gen/dcos-config.yaml]: yes

Comment (ichernetsky) [reply gen/dcos-config.yaml]: How do the variables in this file get available to Marathon then?

**Pull request closed without merging**